### PR TITLE
React goods selector

### PR DESCRIPTION
**Pull request by vsupruniuk**
[DEMO LINK](https://vsupruniuk.github.io/react_goods-selector/)

### CODE REVIEW

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -14,11 +14,98 @@ const goodsFromServer: string[] = [
...
+            return (
+              <li key={good} className={`list-group-item ${isSelected ? 'selectedGood' : ''}`}>
+                <span>
+                  {good}
+                </span>
+
+                <button
+                  type="button"
+                  onClick={() => this.buttonAction(isSelected, good)}
+                  className={`${isSelected ? 'btn btn-danger' : 'btn btn-success'}`}
```
Comment (mgrinko) [inline src/App.tsx:88]: use `classnames` lib here

**Review by mgrinko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +15,114 @@ const goodsFromServer: string[] = [
...
+        return 'No goods selected';
+
+      case 1:
+        return `${selecedGoods[0]} is selected`;
+
+      case 2:
+        return `${selecedGoods[0]} and ${selecedGoods[1]} are selected`;
+
+      default:
+        return `${selecedGoods.slice(0, -1).join(', ')} and ${selecedGoods[selecedGoods.length - 1]} are selected`;
```
Comment (illiarerun) [inline src/App.tsx:64]: `.at` will still be better
```suggestion
        return `${selecedGoods.slice(0, -1).join(', ')} and ${selecedGoods.at(-1)} are selected`;
```

File: src/App.tsx [hunk 3]
```diff
@@ -14,11 +15,114 @@ const goodsFromServer: string[] = [
...
+                  type="button"
+                  onClick={() => this.buttonAction(isSelected, good)}
+                  className={classNames(
+                    'btn',
+                    {
+                      'btn-danger': isSelected,
+                    },
+                    {
+                      'btn-success': !isSelected,
+                    },
```
Comment (illiarerun) [inline src/App.tsx:104]: you can combine these objects
```suggestion
                    {
                      'btn-danger': isSelected,
                      'btn-success': !isSelected,
                    },
```

File: src/App.tsx [hunk 4]
```diff
@@ -14,11 +15,114 @@ const goodsFromServer: string[] = [
...
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selecedGoods: string[],
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selecedGoods: ['Jam'],
```
Comment (illiarerun) [inline src/App.tsx:24]:
```suggestion
    selectedGoods: ['Jam'],
```

**Review by illiarerun — APPROVED** [review]

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**